include/bluetooth/iso.h

```diff
@@ -60,45 +60,43 @@ struct bt_iso_chan {
 	struct bt_iso_chan_ops		*ops;
 	/** Channel QoS reference */
 	struct bt_iso_chan_qos		*qos;
-	/** Channel data path reference*/
-	struct bt_iso_chan_path		*path;
 	sys_snode_t			node;
 	uint8_t				state;
 	bt_security_t			required_sec_level;
 };
 
-/** @brief Audio QoS direction */
-enum {
-	BT_ISO_CHAN_QOS_IN,
-	BT_ISO_CHAN_QOS_OUT,
-	BT_ISO_CHAN_QOS_INOUT
+/** @brief ISO Channel IO QoS structure. */
+struct bt_iso_chan_io_qos {
+	/** Channel interval in us. Value range 0x0000FF - 0x0FFFFFF. */
+	uint32_t			interval;
+	/** Channel Latency in ms. Value range 0x0005 - 0x0FA0. */
+	uint16_t			latency;
+	/** Channel SDU. Value range 0x0000 0 0x0FFF. */
+	uint8_t				sdu;
+	/** Channel PHY - See BT_GAP_LE_PHY for values.
+	 *  Setting BT_GAP_LE_PHY_NONE disables data path.
+	 */
+	uint8_t				phy;
+	/** Channel Retransmission Number. Value range 0x00 - 0x0F. */
+	uint8_t				rtn;
+	/** Channel data path reference.
+	 *  Setting to NULL default to HCI data path.
+	 */
+	struct bt_iso_chan_path		*path;
 };
 
 /** @brief ISO Channel QoS structure. */
 struct bt_iso_chan_qos {
-	/** @brief Channel direction
-	 *
-	 *  Possible values: BT_ISO_CHAN_QOS_IN, BT_ISO_CHAN_QOS_OUT or
-	 *  BT_ISO_CHAN_QOS_INOUT. Shall be BT_ISO_CHAN_QOS_IN for broadcast
-	 *  transmitting, and BT_ISO_CHAN_QOS_OUT for broadcast receiver.
-	 */
-	uint8_t				dir;
-	/** Channel interval in us. Value range 0x0000FF - 0x0FFFFFF. */
-	uint32_t			interval;
 	/** Channel SCA - Only for CIS */
 	uint8_t				sca;
 	/** Channel packing mode. 0 for unpacked, 1 for packed. */
 	uint8_t				packing;
 	/** Channel framing mode. 0 for unframed, 1 for framed. */
 	uint8_t				framing;
-	/** Channel Latency in ms. Value range 0x0005 - 0x0FA0. */
-	uint16_t			latency;
-	/** Channel SDU. Value range 0x0000 0 0x0FFF. */
-	uint8_t				sdu;
-	/** Channel PHY - See BT_GAP_LE_PHY for values. Shall not be BT_GAP_LE_PHY_NONE. */
-	uint8_t				phy;
-	/** Channel Retransmission Number. Value range 0x00 - 0x0F. */
-	uint8_t				rtn;
+	/* Channel Receiving QoS: Data Path BT_HCI_DATAPATH_DIR_CTLR_TO_HOST */
+	struct bt_iso_chan_io_qos	rx;
+	/* Channel Transmission QoS: Data Path BT_HCI_DATAPATH_DIR_CTLR_TO_HOST */
+	struct bt_iso_chan_io_qos	tx;
```

Comment:
I wonder if it would be better to have `rx` and `tx` as pointers to determine IN/OUT or TX/RX, rather than the PHY.

Reply:
Yep, I was thinking that perhaps PHY is not that great, it adds 2 pointers and it yet another thing that the application needs to allocate separetly by it seems much cleaner and perhaps for BIS it actually saves space since we would never have both tx and rx.

Reply:
I was thinking a union might be a solution to avoid separate allocation - but for a bidirectional CIS, won't you have both there?

Reply:
For bidirectional CIS you would have both, so we would have a slightly higher memory usage there, but great savings if using unidirectional. 

I guess it would also technically be possible to reuse the same `bt_iso_chan_io_qos` struct for multiple CIGs or even reuse a tx `bt_iso_chan_io_qos` for both a CIG and a BIG. 

Overall I think this is an improvement.